docs/docsite/rst/dev_guide/debugging.rst

```diff
@@ -1,140 +1,110 @@
+      /home/badger/.ansible/tmp/ansible-tmp-1461434734.35-235318071810595/debug_dir
+
+    If you want to examine the wrapper file you can. It will show a small Python script with a large base64 encoded string. The string contains the module to execute.
+
+#. When you look into the temporary directory you'll see a structure like this:
+
+    .. code-block:: shell-session
+
+      ├── AnsiballZ_ping.py
+      └── debug_dir
+          ├── ansible
+          │   ├── __init__.py
+          │   ├── module_utils
+          │   │   ├── __init__.py
+          │   │   ├── _text.py
+          │   │   ├── basic.py
+          │   │   ├── common
+          │   │   ├── compat
+          │   │   ├── distro
+          │   │   ├── parsing
+          │   │   ├── pycompat24.py
+          │   │   └── six
+          │   └── modules
+          │       ├── __init__.py
+          │       └── ping.py
+          └── args
+
+    * ``AnsiballZ_ping.py`` is the Python script with the the module code stored in a base64 encoded string. It contains various helper functions for executing the module.
+
+    * ``ping.py`` is the code for the module itself. You can modify this code to see what effect it would have on your module, or for debugging purposes.
+
+    * The ``args`` file contains a JSON string. The string is a dictionary containing the module arguments and other variables that Ansible passes into the module to change its behavior. Modify this file to change the parameters passed to the module.
+
+    * The ``ansible`` directory contains the module code in ``modules`` as well as code from :mod:`ansible.module_utils` that is used by the module. Ansible includes files for any :mod:`ansible.module_utils` imports in the module but not any files from any other module. If your module uses :mod:`ansible.module_utils.url` Ansible will include it for you. But if your module includes `requests <https://requests.readthedocs.io/en/master/api/>`_, then you'll have to make sure that the Python `requests library <https://pypi.org/project/requests/>`_ is installed on the system before running the module.
+
+    You can modify files in this directory if you suspect that the module is having a problem in some of this boilerplate code rather than in the module code you have written.
+
+#. Once you edit the code or arguments in the exploded tree, use the ``execute`` subcommand to run it:
+
+    .. code-block:: shell-session
+
+      $ python AnsiballZ_ping.py execute
+      {"invocation": {"module_args": {"data": "debugging_session"}}, "changed": false, "ping": "debugging_session"}
+
+    This subcommand takes care of setting the ``PYTHONPATH`` to use the exploded ``debug_dir/ansible/module_utils`` directory and invoking the script using the arguments in the ``args`` file. You can continue to run the module like this until you understand the problem. Then you can copy the changes back into your real module file and test that the real module works via ``ansible`` or ``ansible-playbook``.
+
```

Comment:
@samdoran Can we add some details about custom PYTHONPATH since collections might be installed at a different locations? Also, adding debbuging with Pycharm and VS code will be good to have but not a blocker.

Reply:
> 
> @samdoran Can we add some details about custom PYTHONPATH since collections might be installed at a different locations? Also, adding debbuging with Pycharm and VS code will be good to have but not a blocker.

POV from documenter
Agree for the PYTHONPATH as it makes sense.
About adding information about different editors I'm a bit reluctant, I fear we will have to maintain that for all editors people want to use, and they may be out of date pretty quickly.  
However that's kind of information that may be interesting in a blog post on the Ansible blog for a particular editor.

Reply:
Make sense.

Reply:
Scratch the idea of Editor specific debugging steps. @bmillemathias Thanks.

Reply:
> Scratch the idea of Editor specific debugging steps. @bmillemathias Thanks.

And I'm the bullhorn team would be more than happy to talk about this future blog post :)

Reply:
@Akasurde I could mention that other debuggers should work with these methods as well. I assumed that was somewhat implied, but that's a bad assumption on my part. @nitzmahone did a Fest talk a while back on using the PyCharm remote debugger, so it is possible.